types/index.d.ts

```diff
@@ -183,6 +187,138 @@ declare module '@azure/functions' {
      * Possible values for an HTTP Request user type
      */
     export type HttpRequestUserType = 'AppService' | 'StaticWebApps';
+    /**
+     * HTTP response object
+     */
+    export interface IResponse {
+        /**
+         * HTTP response headers.
+         */
+        headers?: HttpResponseHeaders;
+        /**
+         *  HTTP response cookies.
+         */
+        cookies?: Cookie[];
+        /**
+         * HTTP response body.
+         */
+        body?: any;
+        /**
+         * HTTP response status code.
+         * @default 200
+         */
+        statusCode?: number | string;
+        /**
+         * Enables content negotiation using ASP.NET core if true
+         * If false, treat response as raw (default)
+         * @default false
+         */
+        enableContentNegotiation?: boolean;
```

Comment:
Honestly, for the sake of completeness, I would argue to keep it. If any user, even if it's 1%, is already using this property in their code (I assume it had a valid use-case, even for node), then attempting to use the new interface would break their existing code. If the description is confusing, then I would opt for having a more accurate description or a less confusing one over removing it from the interface entirely.

Reply:
> I assume it had a valid use-case, even for node

This comment implies it doesn't actually do anything: https://github.com/Azure/azure-functions-nodejs-worker/issues/168#issuecomment-454657844. If you can verify it actually works, then I'm okay including it based on your reasoning

Reply:
This is just a silly example, but I am sure there are other ways it could be used.

Using `enableContentNegotiation: false` (default):

```TS
context.res = {
    status: 200,
    body: "{'hello': 'world', 'test': 123}",
    headers: {
        'content-type': 'application/json'
    },
    enableContentNegotiation: false
};
```

Returns the response body as `{'hello': 'world', 'test': 123}`, which is an invalid `application/json` response.

Using `enableContentNegotiation: true`:

```TS
context.res = {
    status: 200,
    body: "{'hello': 'world', 'test': 123}",
    headers: {
        'content-type': 'application/json'
    },
    enableContentNegotiation: true
};
```

returns the body as `"{'hello': 'world', 'test': 123}"` (added double quotes), which is a valid `application/json` response.